pkg/wamr/Makefile

```diff
@@ -0,0 +1,87 @@
+PKG_NAME=wamr
+PKG_URL=https://github.com/bytecodealliance/wasm-micro-runtime.git
+ifeq ($(PKG_BLEADING),1)
+  PKG_VERSION=main
+else
+  PKG_VERSION=770ca8c90c8803f934bf681cd29949b7fb5291a3
+  #main 2022-01-28
+endif
+PKG_LICENSE=Apache-2.0
+
+PKG_CUSTOM_PREPARED = CHECK_VERSION
+
+include $(RIOTBASE)/pkg/pkg.mk
+
+#less Wall TODO: get things fixed upstream
+CFLAGS :=  $(filter-out -Wcast-align, $(CFLAGS))
+CFLAGS :=  $(filter-out -Werror, $(CFLAGS))
+CFLAGS += -Wno-format
+CFLAGS += -Wno-strict-prototypes
+CFLAGS += -Wno-old-style-definition
+CFLAGS += -Wno-cast-function-type
+
+#Prepapre includes for cmake
+RIOT_INCLUDES = $(filter-out -%,$(subst -I,,$(INCLUDES)))
+
+#translate (CPU_ARCH) to Build Target
+#WAMR_BUILD_TARGET is "X86_32" "AARCH64[sub]", "ARM[sub]",
+#                     "THUMB[sub]", "MIPS" or "XTENSA"
+#no msp430, no AVR support for now
+
+ifeq ($(CPU),native)
+#$(CPU) is defined for every CPU
+#everyone build on x86_32
+  WAMR_BUILD_TARGET = X86_32
```

Comment:
Hmm, this is not necessarily true. E.g. `native` has additional features like SPI on the Raspberry Pi, which is just one famous example of a non x86 based platform.

Reply:
you are right but it is not possible to get the architecture of native in the riot build system at that moment -> i went with most common (just checked if OS_ARCH would do -> no)
maybe i should provide something to set that value manually

Reply:
rechecked OS_ARCH may do